app/views/shared/_footer.html.erb

```diff
@@ -1,9 +1,9 @@
-<footer>
+<%= tag.footer class: {"bg-teal-lightest": Feature.enabled?(:redesign)} do %>
   <div class="max-w-7xl mx-auto py-12 px-4 overflow-hidden sm:px-6 lg:px-8">
     <nav class="mb-6 -mx-5 -mt-2 flex flex-wrap justify-center" aria-label="Language links">
       <% I18n.available_locales.each do |locale| %>
         <div class="px-5 py-2">
-          <%= link_to request.params.merge(locale: resolve_locale(locale)), class: "text-base text-gray-500 hover:text-gray-900", "aria-label": t(".language_name_of_locale", locale: locale) do %>
+          <%= link_to request.params.merge(locale: resolve_locale(locale)), class: {"text-base": true, "text-gray-500 hover:text-gray-900": !Feature.enabled?(:redesign), "text-typography hover:text-navy-mid": Feature.enabled?(:redesign)}, "aria-label": t(".language_name_of_locale", locale: locale) do %>
```

Comment:
We don't need to ever set a value as `true` if we convert this to an array. For example:

```suggestion
          <%= link_to request.params.merge(locale: resolve_locale(locale)), class: ["text-base", "text-gray-500 hover:text-gray-900": !Feature.enabled?(:redesign), "text-typography hover:text-navy-mid": Feature.enabled?(:redesign)], "aria-label": t(".language_name_of_locale", locale: locale) do %>
```

Reply:
Also, there's already a ton of repetition in these links. And the conditional from the redesign is making them a bit hard to reason about.

Let's extract a partial or View Component to use for all of these. I think the only parameters need to be the title, the URL, and if it's an external/internal link.

Reply:
Noted on the array syntax. I had seen that used in Matt Swanson's article on the token helper, but I ran into issues using the syntax in another project when I tried. At the time I thought that it was weird seeing hash keys in an array anyways so I ended up using this approach. I wonder if I was using an old version of Ruby.

Do you know what's going on with this syntax to make it legal? Is the Ruby parser assuming that we're trying to embed a hash inside of the array?

Reply:
>  Is the Ruby parser assuming that we're trying to embed a hash inside of the array?

I believe that is what's going on. The array has two elements here, `"text-base"` and then a hash. I think it has something to do with how the last parameter to a method can be an "options hash". But I don't know for certain! Could be a good blog post.